Transcendence/TransCore/RPGShipBroker.xml

```diff
@@ -1151,6 +1183,51 @@
 					
 					(setq shipList (shuffle shipList))
 
+					;   Ensure that we have enough ships to even consider using randomized ships
+
+					(if (gr MinCount 0)
+						(while (and (ls (count shipList) minCount) (eq outOfRandomShips Nil))
+							(block
+								((extendedShipList (map (rpgCreateShipsToSell sellerObj maxCount fallbackCreateCriteria) shipId (objGetObjByID shipId))))
+								(setq shipList (append shipList extendedShipList))
+								(if (eq (count extendedShipList) 0) (setq outOfRandomShips 1))
+								)
+							)
+						)
+
+					;	If we prioritized filling out minCount first, we can stochastically remove some of the ships by rarity
+					(if prioritizeFill (block
+						(
+							;	Dont remove more than this number of ships
+							(maxRemove (- (count shipList) minCount))
+
+							;	helper boolean set op function to remove elements of A if they are in B
+							(lnkCull (lambda (A B) (enum A lA (enum B lB (if (eq lA lB) (lnkRemove A (find A lA)))))))
+
+							;	the actual ships to remove
+							(shipsToRemove (map shipList 'excludeNil theShip (if
+								;	roll against rarity ratio to remove the ship
+								(gr (random) (typGetProperty (objGetType theShip) 'frequencyRatio))
+									shipObj
+
+									;otherwise ignore it
+									Nil
+								)))
+							)
+						
+						;shuffle the marked ships to remove
+						(setq shipsToRemove (shuffle shipsToRemove))
+
+						;only remove up to the maximum number that can be removed
+						(setq shipsToRemove (subset shipsToRemove 0 maxRemove))
+
+						;remove these from shipList
+						(setq shipList (lnkCull shipList shipsToRemove))
```

Comment:
Using the lnkCull lambda is unnecessary as the built in `filter` or `map` functions can be used for this type of set operation:
```suggestion
						(setq shipList (filter shipList theShip (not (find shipsToRemove theShip))))
```

Also it looks like the current lnkCull function is not suitable for use in a setq like this as lnkCull does an in place modification of list A and will only return the modified list IF the final operation removed an element (i.e. last element of shipList and shipsToRemove are the same). If the last elements of shipList and shipsToRemove are different then the lambda will return Nil overwriting the modified shipList